pkg/server/grpc/tracee.go

```diff
@@ -254,14 +260,15 @@ func getK8s(e trace.Event) *pb.K8S {
 	}
 }
 
-func getThreat(e trace.Event) *pb.Threat {
-	if e.Metadata == nil || e.Metadata.Properties == nil {
+func getThreat(description string, metadata map[string]interface{}) *pb.Threat {
+	if metadata == nil {
 		return nil
 	}
 	// if metadata doesn't contain severity, it's not a threat,
-	// severity is set when we have a finding from the signature engine
+	// severity is set when we have an event created from a signature
 	// pkg/ebpf/fiding.go
-	_, ok := e.Metadata.Properties["Severity"]
+	// pkg/cmd/initialize/sigs.go
+	_, ok := metadata["Severity"]
```

Comment:
The code to convert types.Event to api.Event is temporary, it all be removed soon, because the event protobuf will  be generated from inside tracee, no conversion required, that is why the code here is ugly.

Reply:
The protobuf will be tracee internal format. But the metadata will depend more on the signatures created by research than tracee itself. Tracee will only load the plugin.